src/readbin/fsi_benchmark.ml

```diff
@@ -0,0 +1,112 @@
+open Core_kernel.Std
+open Cmdliner
+open Bap.Std
+open Bap_plugins.Std
+
+type gt_format = [ 
+  | `unstripped_bin
+  | `symbol_file ]
+
+type result = {
+  tool_name: string;
+  tp: int;
+  fn: int;
+  fp: int;
+  prec: float;
+  recl: float;
+  f_05: float}
+
+let tool : string Term.t =
+  let doc = "The tool of the function start result that we are going to
+  evaluate. If user wants to evaluate bap-byteweight, one can use
+  \"bap-byteweight\". If user want to evaluate IDA, one can use
+  \"idaq\", \"idal\", \"idaq64\" or the specific path of IDA." in
+  Arg.(required & pos 0 (some string) (Some "bap-byteweight") & info [] ~docv:"tool" ~doc)
+
+let bin : string Term.t =
+  let doc = "The testing stripped binary." in
+  Arg.(required & pos 1 (some non_dir_file) None & info [] ~docv:"binary" ~doc)
+ 
+let gt : string Term.t =
+  let doc =
+    "The ground truth. The ground truth can be an unstripped
+  binary, or a .scm file with symbol information. In .scm file, each symbol should
+  be in format of:
+  (<symbol name> <symbol start address> <symbol end address>), e.g.
+  (malloc@@GLIBC_2.4 0x11034 0x11038)" in
+  Arg.(required & pos 2 (some non_dir_file) None
+       & info [] ~docv:"ground-truth" ~doc)
+
+let print_metrics : _ list Term.t =
+  let opts = [
+    "precision", `with_prec;
+    "recall", `with_recl;
+    "F_measure", `with_F;
+    "TP", `with_TP;
+    "FP", `with_FP;
+    "FN", `with_FN;
+  ] in
+  let doc = "Print metrics. User can choose to print -cprecision, -crecall,
+  -cF_measure, -cTP, -cFP and -cFN." in
+  Arg.(value & opt_all (enum opts) (List.map ~f:snd opts) &
+       info ["with-metrics"; "c"] ~doc)
+
+let print {tool_name;fp;fn;tp;prec;recl;f_05} print_metrics =
```

Comment:
Instead of accumulating a list of strings for the later printing just print them directly. 
Write a function `string_of_metric` and use it to output header with metrics first.
Write a function `output_metric_value : out_channel -> result -> metric`  and later `List.iter` over the metrics with this function.